smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddHttpApiKeyAuthPlugin.java

```diff
@@ -0,0 +1,178 @@
+            RuntimeClientPlugin.builder()
+                .inputConfig(Symbol.builder().namespace(
+                    "./" + CodegenUtils.SOURCE_FOLDER + "/middleware/HttpApiKeyAuth", "/"
+                ).name("HttpApiKeyAuthInputConfig").build())
+                .resolvedConfig(Symbol.builder().namespace(
+                    "./" + CodegenUtils.SOURCE_FOLDER + "/middleware/HttpApiKeyAuth", "/"
+                ).name("HttpApiKeyAuthResolvedConfig").build())
+                .resolveFunction(Symbol.builder().namespace(
+                    "./" + CodegenUtils.SOURCE_FOLDER + "/middleware/HttpApiKeyAuth", "/"
+                ).name("resolveHttpApiKeyAuthConfig").build())
+                .servicePredicate((m, s) -> hasHttpApiKeyAuthTrait(s) && !areAllOptionalAuthOperations(m, s))
+                .build(),
+
+            // Add the middleware to operations that use HTTP API key authorization
+            RuntimeClientPlugin.builder()
+                .pluginFunction(Symbol.builder()
+                    .namespace("./" + CodegenUtils.SOURCE_FOLDER + "/middleware/HttpApiKeyAuth", "/")
+                    .name("getHttpApiKeyAuthPlugin")
+                    .build())
+                .additionalPluginFunctionParamsSupplier((m, s, o) -> new HashMap<String, Object>() {{
+                    // It's safe to do getTrait().get() because the operation predicate ensures that the trait exists
+                    // `in` and `name` are required attributes of the trait, `scheme` is optional
+                    put("in", s.getTrait(HttpApiKeyAuthTrait.class).get().getIn().toString());
+                    put("name", s.getTrait(HttpApiKeyAuthTrait.class).get().getName());
+                    put("scheme", s.getTrait(HttpApiKeyAuthTrait.class).get().getScheme().orElse(null));
+                }})
+                .operationPredicate((m, s, o) -> hasHttpApiKeyAuthTrait(s)
+                                                    && !operationUsesOptionalAuth(m, s, o))
+                .build()
+        );
+    }
+
+    @Override
+    public void writeAdditionalFiles(
+        TypeScriptSettings settings,
+        Model model,
+        SymbolProvider symbolProvider,
+        BiConsumer<String, Consumer<TypeScriptWriter>> writerFactory
+    ) {
+        ServiceShape service = settings.getService(model);
+
+        // If the service doesn't use HTTP API keys, we don't need to do anything and the generated
+        // code doesn't need any additional files.
+        if (!hasHttpApiKeyAuthTrait(service) || areAllOptionalAuthOperations(model, service)) {
+            return;
+        }
+
+        String noTouchNoticePrefix = "// Please do not touch this file. It's generated from a template in:\n"
+                + "// https://github.com/awslabs/smithy-typescript/blob/main/smithy-typescript-codegen/"
+                + "src/main/resources/software/amazon/smithy/aws/typescript/codegen/integration/";
+
+        // write the middleware source
```

Comment:
nit: comments should be complete sentences, starting with capitals and ending with periods.

This occurs in other places as well (including the test file.)